solr/solrj/src/java/org/apache/solr/client/solrj/util/AsyncListener.java

```diff
@@ -19,10 +19,23 @@
 
 /** Listener for async requests */
 public interface AsyncListener<T> {
-  /** Callback method invoked before processing the request */
+  /**
+   * Setup method invoked in the original thread before processing the request on a different
```

Comment:
💯 very important clarification that explains why we're later getting the right `SolrRequestInfo`.

Reply:
yeah got very scared initially when I saw SolrRequestInfo referenced in an implementation of AsyncListner